### PR TITLE
User Visibility by Role Preferences

### PR DESCRIPTION
**Pull request by micahmills**
Adds a Preference pane in Settings (DT) -> General Settings. This lists all the DT Roles and allows you to check the box if you want to allow that role to have access to the DT users on the front end. This will allow things like Assign, Sub-assign, and @ mentions to list all DT Users.

### CODE REVIEW

**Comment by micahmills** [comment]
I am not 100% happy with the UI for the preference pane. It might need to be a little more clear, but this is functional.

**Comment by corsacca** [comment]
Users were showing up in the groups typeahead on the contact record.

I'm not comfortable with multipliers with the `list_users` permission having access to the `/user-management/users/` page.

**Comment by micahmills** [comment]
I didn't realize that they had that access to the /user-management/users I think that would probably be solved by adding a custom capability instead of the list_users.

File: dt-users/users.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public static function get_assignable_users_compact( string $search_string = nul
         $user_id = get_current_user_id();
         $users = [];
         $update_needed = [];
-        if ( !user_can( get_current_user_id(), 'view_any_contacts' ) ){
+        if ( !user_can( get_current_user_id(), 'view_any_contacts' ) && !user_can( get_current_user_id(), 'list_users' ) ){
```
Comment (corsacca) [inline dt-users/users.php:69]: dt_list_users

**Comment by corsacca** [comment]
For clarity: can we `disable` the rows that have `view_any_contacts` permission. I'm not sure we wan't to disable "list users" for those roles, and it does not look like unchecking them removes the capacity to list users..
image

**Comment by micahmills** [comment]
Yes that makes sense

**Comment by micahmills** [comment]
Having the Admin and other roles with view_any_contacts capabilities listed but disabled felt awkward so I just removed them from the list. Now the list only have User roles that don't inherently have the ability to view all contacts.

**Comment by corsacca** [comment]
Thanks @micahmills !

**Pull request merged**